### PR TITLE
Add `SyncedPlayerChanged`

### PR DESCRIPTION
**Pull request by sprunk**
Except for demo-watching specs, when these do something it's still unsynced only

### CODE REVIEW

**Comment by rtri** [comment]
This is moving in the wrong direction for my taste, I'd add a SyncedPlayerChanged event.

**Comment by sprunk** [comment]
Alright.

**Comment by abma** [comment]
@sprunk:

as i only partly understand this change: what problem does this solve?

**Comment by sprunk** [comment]
Synced currently doesn't know about the PlayerChanged event, so you can't do things when somebody disconnects or reconnects (for example a gadget that gives the abandoned units so somebody can control them, and then returns them on reconnect). I believe it is unavailable because people watching a replay can also generate the event, which is why the patch makes a new callin (cutting replay-watching spectators from regular PlayerChanged would hurt unsynced gadgets and LuaUI, and exposing them to synced would open up the possibility of a desync).

**Comment by abma** [comment]
ugh, the merge is wrong. i'll do in cli.

thanks for explaining!

**Comment by sprunk** [comment]
I copied the explanation to the code as a comment: 747b49cd6bedc672283b3fd691b6e8143e8cd5bb

File: rts/Net/NetCommands.cpp [hunk 1]
```diff
@@ -1291,7 +1291,20 @@ void CGame::ClientReadNet()
 						skirmishAIHandler.RemoveSkirmishAI(aiNum);
 
 						CPlayer::UpdateControlledTeams();
-						eventHandler.PlayerChanged(playerNum);
+						playerHandler.Player(playerNum)->NotifyPlayerChanged();
+
+						LOG("Skirmish AI \"%s\" (ID:%i), which controlled team %i is now dead", aiInstanceName.c_str(), aiNum, aiTeamId);
```
Comment (abma) [inline rts/Net/NetCommands.cpp:1296]: rts/Net/NetCommands.cpp:1296:79: error: use of undeclared identifier 'aiInstanceName'

  ...AI \"%s\" (ID:%i), which controlled team %i is now dead", aiInstanceName...

File: doc/changelog.txt [hunk 2]
```diff
@@ -150,6 +151,8 @@ Lua:
  - MarkerAdd{Point,Line} now have an extra optional argument (after localOnly: 6th for point, 8th for line):
    the playerID of the player who should appear as the author of the drawing (if missing, defaults to the local player).
    Only has an effect when localOnly is true.
+
+=======
```
Comment (abma) [inline doc/changelog.txt:155]: my fault: should be removed

**Review by abma — CHANGES_REQUESTED** [review]

**Comment by abma** [comment]
https://github.com/spring/spring/commit/aa7a9972a26c15cb41d9d438fc457a86d58f7630

**Pull request closed without merging**